migrate/20241108_resource_group.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+Sequel.migration do
+  change do
+    create_enum(:resource_group_type, %w[dedicated shared])
+
+    create_table(:resource_group) do
+      column :id, :uuid, primary_key: true
+      column :name, :text, null: false
+      column :allocation_state, :allocation_state, default: "unprepared", null: false
```

Comment:
I'm a bit unsure about if we should use the `allocation_state` enum here, as it's tied a bit to tracking state of a VmHost. Although right now this seems fine, but using it for a 2nd purpose might make changing it more difficult in the future.

Can we use an `enabled` boolean instead?

Reply:
OK, fair. I just wanted to not invent a new enum. I can change it to boolean and we will see if we need more than 2 states and come back to it, if needed.